### ISSUE TITLE
Get only the most recently uploaded CSV file from the SFTP server

### ISSUE THREAD

**Issue by aleksa-krolls** [OpenFn/women-for-women#112]
## Background, context, and business value
We have existing flow jobs that (1) get specific CSV files from an SFTP server, convert to JSON, and send to Inbox as Messages... then (2) map and load that data to Salesforce. This workflow currently runs on a scheduled weekly basis, and gets whichever file name is specified in the job. 

## The specific request, in as few words as possible
The client would like to start doing daily data syncs, so we will change the workflow trigger to run daily @ 12:00 AM. 

When the `Get CSVs` job runs, we'd like to dynamically fetch only the most recently uploaded file from the SFTP server. There may be several files with the same name, but the date `YYYYMMDD` will differ in the file name (e.g., `wfwi Donors 20230222.csv`).

For example, if in the SFTP server we find the following files... 
```
wfwi Donors 20230221.csv
wfwi Donors 20230222.csv
```
Only fetch the most recent file: `wfwi Donors 20230222.csv` because Feb 22nd is more recent than Feb 21st. 

## state.json
You'll find a couple of `wfwi Donors` files currently on this SFTP server... see if you can modify the job to fetch only the most recent file `wfwi Donors 20230222.csv`. 

```json
{
  "configuration": { SEE LP `WFWI SFTP` for the config },
  "data": { }
}
```

## adaptor
`sftp`

## expression.js
https://github.com/OpenFn/women-for-women/blob/master/jobs/committed/1.fetchJSON_Donors.js

## output.json
If you successfully get the file from the SFTP server, it will send a Message to [this project inbox](https://www.openfn.org/projects/wfwi-committed-giving/messages?dateRange=720&endDate&jobStates=&page=1&perPage=10&searchString=&startDate=2023-01-24T07%3A37%3A48.108Z&triggerId=). 

## Toggl
`WFWI Support 2023`

**Comment by aleksa-krolls** [comment]
@mtuchi this is another issue that is ready to be picked up today. Ping me if any questions.

**Comment by mtuchi** [comment]
> @mtuchi this is another issue that is ready to be picked up today. Ping me if any questions.

@aleksa-krolls the PR is ready for QA and testing see 👉🏽  #113

**Comment by aleksa-krolls** [comment]
hey @mtuchi when I test this on platform, the job is getting ALL csvs with the most recent date - uploaded Feb 18th: `20230228`
[Check out the inbox](https://www.openfn.org/projects/wfwi-committed-giving/messages?dateRange=720&endDate&jobStates=&page=1&perPage=10&searchString=&startDate=2023-01-30T06%3A40%3A15.413Z&triggerId=) where you'll see several Messages were created today, and include data from other files like `wfwi Transactions - Cash 20230228.csv` as well. 

I only want to return the most recent `wfwi donors` file --> so if you run this job today on the sftp server, it should only return the one file `wfwi Donors 20230228.csv`

**Comment by mtuchi** [comment]
> hey @mtuchi when I test this on platform, the job is getting ALL csvs with the most recent date - uploaded Feb 18th: `20230228` [Check out the inbox](https://www.openfn.org/projects/wfwi-committed-giving/messages?dateRange=720&endDate&jobStates=&page=1&perPage=10&searchString=&startDate=2023-01-30T06%3A40%3A15.413Z&triggerId=) where you'll see several Messages were created today, and include data from other files like `wfwi Transactions - Cash 20230228.csv` as well.
> 
> I only want to return the most recent `wfwi donors` file --> so if you run this job today on the sftp server, it should only return the one file `wfwi Donors 20230228.csv`

@aleksa-krolls I have merged a fix (#114), please test now on platform-app
